functions/go/source-gcloud-config/main.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/source-gcloud-generator/generated"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework/command"
+)
+
+type Processor struct{}
+
+func (p *Processor) Process(resourceList *framework.ResourceList) error {
+	err := func() error {
+		gen := &GcloudConfigGenerator{}
```

Comment:
We want to vendor this function in `kpt` repo as builtin. To be able to invoke it as library, we will have to refactor it so that the core logic can be executed by CLI (like here) and also from the kpt code where it accepts serialized input resource-list and output serialized resource-list.

@martinmaly had an example for that in apply-setters, not sure if that got merged.

Reply:
Done.  the lib would be like 
```go
import "github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/source-gcloud-generator/generator"

func () {
       gen  := generator.NewGcloudConfigGenerator()
       newItems, err := gen.Generate(resourceList.Items)
       ...  
}
```